packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol

```diff
@@ -190,6 +191,24 @@ contract ZoraCreator1155PremintExecutorImpl is
         );
     }
 
+    /// @notice Returns the version of the premint signature that the contract supports
+    /// @param contractAddress The address of the contract to check
+    /// @return The version of the premint signature that the contract supports.  If it doesn't support premint
+    /// returns 0
+    function supportedPremintSignatureVersion(address contractAddress) external view returns (string memory) {
+        IZoraCreator1155 creatorContract = IZoraCreator1155(contractAddress);
+        if (creatorContract.supportsInterface(type(IZoraCreator1155DelegatedCreation).interfaceId)) {
+            return IZoraCreator1155DelegatedCreation(contractAddress).supportedPremintSignatureVersion();
+        }
+
+        // try get token id for uid 0 - if call fails, we know this didn't support premint
+        try ERC1155DelegationStorageV1(contractAddress).delegatedTokenId(uint32(0)) returns (uint256) {
```

Comment:
are we somehow ensuring that the first premint executed on a contract is uid 0?

Reply:
it doesn't really matter here - this is really just checking if that function exists on the contract. even if the uid is not 0, this will return 0.  It will revert if the function doesn't exist on the contract